File: 283-move-zeroes/283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        list_pointer: int = 0;
        zero_counter: int = 0;
        
        for index, value in enumerate(nums):
            if value == 0:
                zero_counter += 1
            else:
                nums[list_pointer] = nums[index]
                list_pointer += 1
        for zero in range(zero_counter):
            nums[list_pointer] = 0
            list_pointer += 1
            

File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int>::iterator it;
        vector<int> zeroLocation = {};
        
        for(int i=0; i<nums.size(); i++){
             if(nums[i] == 0){
                 zeroLocation.push_back(i);
             }
        }
        for(int i=zeroLocation.size()-1; i>=0; i--){
            it=nums.begin()+zeroLocation[i];
            nums.erase(it);
            nums.push_back(0);
        }
    }
};